arcgis-ios-sdk-samples/Layers/Raster rendering rule/RasterRenderingRuleViewController.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class RasterRenderingRuleViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The choose rendering rule button.
+    @IBOutlet var chooseRenderingRuleBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties and methods
+    
+    /// An online image service that features Charlotte, North Carolina's downtown area.
+    let imageServiceURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer")!
+    /// A list of rendering rule info supported by the service.
+    var rasterRenderingRules = [AGSRenderingRuleInfo]()
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .streets())
+        // Create a raster layer from the raster and add it to the map.
+        let imageServiceRaster = AGSImageServiceRaster(url: imageServiceURL)
+        map.operationalLayers.add(makeRasterLayer(raster: imageServiceRaster))
+        // Load the raster and get a list of rendering rule info supported by the service.
+        imageServiceRaster.load { [weak self] error in
+            guard let self = self else { return }
+            if let serviceInfo = imageServiceRaster.serviceInfo, let extent = serviceInfo.fullExtent {
+                self.mapView.setViewpoint(AGSViewpoint(targetExtent: extent), completion: nil)
+                self.rasterRenderingRules = serviceInfo.renderingRuleInfos
+                self.chooseRenderingRuleBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
```

Comment:
I think we generally keep the error handling at the beginning of the completion handler but I don't see an issue with how it is now. Just a thought!

Reply:
I cannot find the original post, but I think Phil suggested otherwise. Let's wait for Phil's call.

Reply:
@vquach2404 We sometimes need to put the error check first when that is the only way to know whether the operation failed. But when possible, I think it makes more sense to put the most likely (no error) case first.

Reply:
Thanks for clarification. I've documented this in the coding style section in our wiki. (/common-samples/wiki/iOS-Samples-Repo:-A-Contribution-Guide#first-success-then-failure)